### PR TITLE
[Home Page Picker] Modify site creation flow to support creating sites by selecting a design

### PR DESCRIPTION
**Pull request by chipsnyder**
### Description
- Expands the site creation endpoint to accept the `template` options so now sites can be created based on a site design instead of a segment
- Updates the `nux/starter-designs` endpoint to include `language` which is needed instead of `locale` for translations.
- Adds support for the field for `is_blog` to support the client in determining if this supports .blog domains or not.

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2721
**Related PR**
- `WordPress-iOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/15231

### Testing Details

This can be tested with the build in the PR https://github.com/wordpress-mobile/WordPress-iOS/pull/15231

<details>
<summary>Tests from related PR</summary>

## To test:
<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Site Creation: Home Page Picker" to enable or disable the picker
        - <kdb><a href="https://user-images.githubusercontent.com/3384451/97343011-5f032980-185d-11eb-9030-06c17f732fec.gif"></a></kdb>

</details>

When enabled, the first step of the Site Creation flow will be replaced with the Home Page Picker flow.

### Create a Site with Site Design
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Select a design
1. Tap on "Choose"
1. Search for and Choose a domain
    - **Expect** to see `.wordpress.com` subdomains only unless you selected an option from the test `Create a Blog`
1. Tap "Create Site"

**Expect** Your site to be created

### Create a Site by skipping site design
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Select "Skip"
1. Search for and Choose a domain
1. Tap "Create Site"
**Expect** Your site to be created

### Create a Blog
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Select a design that has been flagged as a blog
    <details>
    <summary>Examples</summary>
    <kdb><a href="https://user-images.githubusercontent.com/3384451/98397556-87451200-202d-11eb-92f6-c42e86a1a7d2.png"></a></kdb>
    </details>
1. Tap on "Choose"
1. Search for and Choose a domain
    - **Expect** to see `.wordpress.com` and `.blog` subdomains.
1. Tap "Create Site"
**Expect** Your site to be created

## Validate existing flow:

To make sure nothing is broken in the existing flows (in case we want to preserve it for A/B test scenarios). Disable the Home page picker feature. 

### Create a Site
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Select a segment other than Blog
1. Search for and Choose a domain
    - **Expect** to see `.wordpress.com` subdomains only.
1. Tap "Create Site"
**Expect** Your site to be created

### Create a Blog
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Select the Blog segment
1. Search for and Choose a domain
    - **Expect** to see `.wordpress.com` and `.blog` subdomains.

</details>

- [x] Please check here if your pull request includes additional test coverage.

### CODE REVIEW

File: WordPressKit/RemoteSiteDesign.swift [hunk 1]
```diff
@@ -22,5 +24,6 @@ public struct RemoteSiteDesign: Codable {
         demoURL = try map.decode(String.self, forKey: .demoURL)
         screenshot = try? map.decode(String.self, forKey: .screenshot)
         themeSlug = try? map.decode(String.self, forKey: .themeSlug)
+        isBlog = (try? map.decode(Bool.self, forKey: .isBlog)) ?? false
```
Comment (chipsnyder) [inline WordPressKit/RemoteSiteDesign.swift:27]: I'm not 100% sure about how we'll make this distinction but went ahead and added one path here. Worst-case scenario, `isBlog` can become a computed property based on whatever values are supplied by the service.

**Comment by chipsnyder** [comment]
Hey @antonis and @mkevins I have this pretty much ready but there are a few test cases here that we can't fully validate until some other work is done. But if you'd like to take an initial look then this is basically good to go.

**Review by antonis — APPROVED** [review]
Nice work Chip 👍 
The code LGTM thought I cannot fully test it till the server response includes the `segment_id`

**Comment by chipsnyder** [comment]
@antonis @mkevins `segement_id` is now available on the backend so this piece is good to go for an official review 👍

File: WordPressKitTests/SiteCreationRequestEncodingTests.swift [hunk 2]
```diff
@@ -3,377 +3,186 @@ import XCTest
...
-        let actualVerticalId = actualOptions!["site_vertical"] as? String
-        XCTAssertNotNil(actualVerticalId)
-        XCTAssertEqual(expectedVerticalId, actualVerticalId!)
-
-        let actualSiteInfo = actualOptions!["site_information"] as? [String : AnyObject]
-        XCTAssertNotNil(actualSiteInfo)
-
-        let actualTagline = actualSiteInfo!["site_tagline"] as? String
-        XCTAssertNotNil(actualTagline)
-        XCTAssertEqual(expectedTagline, actualTagline!)
```
Comment (mkevins) [inline WordPressKitTests/SiteCreationRequestEncodingTests.swift:90]: :heart: the refactor extracting these to `validate` :smile:

**Review by mkevins — APPROVED** [review]
Tested via https://github.com/wordpress-mobile/WordPress-iOS/pull/15231 and everything works as expected. LGTM!

**Pull request merged**